pkg/provider/azure_standard.go

```diff
@@ -1142,7 +1142,8 @@ func (as *availabilitySet) EnsureBackendPoolDeleted(service *v1.Service, backend
 			ipconfigPrefixToNicMap[ipConfigIDPrefix] = nic
 		}
 	}
-	for _, nic := range ipconfigPrefixToNicMap {
+	for k := range ipconfigPrefixToNicMap {
```

Comment:
Good catch. Is there any golint rules could be enabled to catch such issues?

Reply:
I think we should have it: golangci-lint but somehow it isn't working here. I'll investigate how to solve it.

Reply:
that's a classic pointer bug in go range statement, it's interesting how to catch it by some scanning tool.